client/app.jsx

```diff
@@ -1,57 +1,40 @@
 import React from 'react';
-import ReactDOM from 'react-dom'; 
-import Form from './form';
-import Nav from './nav';
-import Login from './login'; 
+import { render } from 'react-dom'; 
+import { Router, Route, hashHistory } from 'react-router';
 
-import { Button } from 'react-bootstrap';
-import $ from 'jquery'; 
+import Login from './login'; 
+import Main from './main';
+import Feed from './feed';
+import Form from './form';
+// import { Button } from 'react-bootstrap';
+// import $ from 'jquery'; 
 
 var sampleData = [{
-	url:"http://assets.fodors.com/destinations/54494/alamo-square-san-francisco-california-usa_main.jpg",
+  thumbnail:"http://assets.fodors.com/destinations/54494/alamo-square-san-francisco-california-usa_main.jpg",
+  src:"http://assets.fodors.com/destinations/54494/alamo-square-san-francisco-california-usa_main.jpg"
 }, {
-	url: "https://pixabay.com/static/uploads/photo/2015/03/26/09/48/golden-gate-bridge-690346_960_720.jpg",
+  thumbnail: "https://pixabay.com/static/uploads/photo/2015/03/26/09/48/golden-gate-bridge-690346_960_720.jpg",
+  src: "https://pixabay.com/static/uploads/photo/2015/03/26/09/48/golden-gate-bridge-690346_960_720.jpg"
 }, {
-	url: "http://www.wheretraveler.com/sites/default/files/styles/main_slider/public/San-Francisco-shutterstock_121582312.jpg?itok=sGTj8sv2", 
+  thumbnail: "http://www.wheretraveler.com/sites/default/files/styles/main_slider/public/San-Francisco-shutterstock_121582312.jpg?itok=sGTj8sv2", 
+  src: "http://www.wheretraveler.com/sites/default/files/styles/main_slider/public/San-Francisco-shutterstock_121582312.jpg?itok=sGTj8sv2" 
 }, {
-	url: "http://www.dog-learn.com/dog-breeds/pomeranian/images/pomeranian-u6.jpg"
+  thumbnail: "http://www.dog-learn.com/dog-breeds/pomeranian/images/pomeranian-u6.jpg",
+  src: "http://www.dog-learn.com/dog-breeds/pomeranian/images/pomeranian-u6.jpg"
 }, {
-	url: "http://animalsbreeds.com/wp-content/uploads/2014/11/Pomeranian-11.jpg"
+  thumbnail: "http://animalsbreeds.com/wp-content/uploads/2014/11/Pomeranian-11.jpg",
+  src: "http://animalsbreeds.com/wp-content/uploads/2014/11/Pomeranian-11.jpg"
 }];
 
-class Main extends React.Component {
-	constructor (props) {
-		super(props);
-		this.submitHandler = this.submitHandler.bind(this); 
-	}
-	
-	submitHandler (startDate, endDate, options) {
-		$.post({
-			url: '/create',
-			data: {
-				startDate: startDate,
-				endDate: endDate,
-				options: options
-			},
-			success: function() {
-				console.log('success'); 
-			}
-		})
-	}
-
-	render () {
-		return (
-			<div>
-				<header>
-					<Nav />
-				</header>
-				<h1>Formidable Fotos</h1>
-				<Login />
-				<Form createNew={this.submitHandler} />
-			</div>
-		); 
-	}
-}; 
 
-ReactDOM.render(<Main data={sampleData} />, document.getElementsByClassName('signIn')[0]);
+render((  
+	<Router history={hashHistory}>
```

Comment:
No I actually didn't see anything on the difference. Now that you mention it though, it looks like even the React Router tutorial bounces back and forth. For example, they used the browserHistory in their "Navigating Programmatically" module. However I just verified it will work either way as long as you're consistent about 1 vs the other. https://github.com/reactjs/react-router-tutorial/tree/master/lessons/12-navigating